### ISSUE TITLE
Resolve error in node 17; DeprecationWarning: Use of deprecated folder mapping in node 16

### ISSUE THREAD

**Issue by daKmoR** [lit/lit#2409]
# Description

Using node 16+ you get the following deprecation message: 

> (node:31725) [DEP0148] DeprecationWarning: Use of deprecated folder mapping "./directives/" in the "exports" field module resolution of the package at /[...]/node_modules/lit/package.json.
Update this package.json to use a subpath pattern like "./directives/*".

Using node 17+ you hit the following error:

> Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './directives/async-append.js' is not defined by "exports" in /[...]/lit/package.json

# Steps to Reproduce

1. Write this code

```ts
import { until } from 'lit/directives/until.js';
```

## Expected Results

Executes in node 17
No DeprecationWarning in node 16

## Actual Results

Fails in node 17
Has DeprecationWarning in node 16

## Additional Info

See the official docs with deprecation
https://nodejs.org/dist/latest-v16.x/docs/api/packages.html#subpath-folder-mappings

Subpath patterns did not exist when lit was initially released so it used the subpath mapping and enforced `*.js` everywhere.

With subpath patterns this may not be necessary anymore.

👉 `package.json`
```
  "exports": {
    "./directives/": {
      "default": "./directives/*.js"
    },
  }
```

allows imports like

```ts
import { until } from 'lit/directives/until';
```

as the subpath pattern will add the `.js`.

However, as this would be a breaking change it's probably easier to still require the `*.js`

👉 `package.json`
```
  "exports": {
    "./directives/": {
      "default": "./directives/*"
    },
  }
```

**TypeScript Note**

As TypeScript still [does not support export maps](https://github.com/microsoft/TypeScript/issues/46452#issuecomment-978371550) I'm not sure if there will be any complications for it 🙈

PS: original [TypeScript Issue](https://github.com/microsoft/TypeScript/issues/33079)

**Comment by stephantabor** [comment]
This is also an issue with Webpack 5 that we are running into with our component library. 

We are building components with typescript, have a `base element` that we extend and something like `export * from 'lit/decorators';`

This is fine in our Typescript source code as it doesn't care about the export declarations in package.json. However when consuming the components with Webpack 5 (default in Create React App) it uses the exports declarations in package.json and throws errors. 

Here's a reproduction: https://github.com/stephantabor/lit-cra-repro

For anyone else that runs into this, the quick and dirty fix I am using is to just delete the `exports` completely from `lit`'s package.json

**Comment by mhoad** [comment]
I believe the original TypeScript problem mentioned has been resolved as of v4.7 https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#package-json-exports-imports-and-self-referencing

**Comment by justinfagnani** [comment]
This should be fixed as we don't use subpath mappings anymore.
